### PR TITLE
use typescript

### LINKED ISSUE #127: use TypeScript

### PR DESCRIPTION
**Pull request by shoota**
Closes #127 

- [x] construct typescript env
- [ ] migrate to ts
- [ ] setup coverage
- [ ] document, and more...

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "tslint-config-standard"
```
Comment (9renpoto) [inline tslint.json]: My company is `@uncovertruth/tslint-config`  host myself, extends standard linter

File: webpack.config.js [hunk 2]
```diff
@@ -16,8 +16,14 @@ module.exports = {
     filename: '[name].js',
     sourceMapFilename: '[name].js.map'
   },
+  resolve: {
+    extensions: ['.ts', '.tsx', '.js']
+  },
   devtool: 'cheap-source-map',
   module: {
+    loaders: [
```
Comment (9renpoto) [inline webpack.config.js]: loaders prefix is old settings

File: src/core.ts [hunk 3]
```diff
@@ -163,7 +170,7 @@ export default class AgentCore extends Store {
 
     if (this.active) {
       const delay = this._interval.shift()
-      if (delay >= 0) {
+      if (delay && delay >= 0) {
```
Comment (9renpoto) [inline src/core.ts]: `through when 0`

File: src/core.ts [hunk 4]
```diff
@@ -1,8 +1,7 @@
-/* @flow */
-import EventEmitter from 'events'
+import * as events from 'events'
```
Comment (9renpoto) [inline src/core.ts]: Why import events all ?

File: src/core.ts [hunk 5]
```diff
@@ -13,44 +12,49 @@ import {
...
   if (c) {
     return c
   }
-  return save(cookieName, generateId(), cookieExpires)
+  return autoCookie.save(cookieName, generateId(), options)
 }
 
-function cacheValidator ({ x, y, type, left, top }: Object): boolean {
-  if (x > 0 && y > 0 && type && left >= 0 && top >= 0) {
+function cacheValidator (c: Interact): boolean {
```
Comment (9renpoto) [inline src/core.ts]: why is not `({ x, y, type, left, top }: Interact)` ?

File: src/events.ts [hunk 6]
```diff
@@ -32,7 +29,7 @@ export default class Events {
     raise('please override validate')
     return false
   }
-  emit (data: { x: number, y: number }): void {
+  emit (data: { x: number; y: number }): void {
```
Comment (9renpoto) [inline src/events.ts]: is this pass ?

Comment (shoota) [reply src/events.ts]: it is written by prettier.

**Comment by shoota** [comment]
Sorry, This PR is too ugly log.  I will do all over again.

**Pull request closed without merging**